client.go

```diff
@@ -303,40 +305,71 @@ func (c *Client) Schema() (*Schema, error) {
 
 // ImportField imports records from the given iterator.
 func (c *Client) ImportField(field *Field, iterator RecordIterator, options ...ImportOption) error {
-	importOptions := &ImportOptions{}
-	if field.options != nil && field.options.fieldType == FieldTypeInt {
-		importRecordsFunction(c.importValues)(importOptions)
-	} else {
-		// Check whether roaring imports is available
-		importOptions.roaring = c.hasRoaringImportSupport(field)
-		importRecordsFunction(c.importColumns)(importOptions)
-	}
+	importOptions := ImportOptions{}
 	for _, option := range options {
-		if err := option(importOptions); err != nil {
+		if err := option(&importOptions); err != nil {
 			return err
 		}
 	}
-	return c.importManager.Run(field, iterator, importOptions.withDefaults())
+	importOptions = importOptions.withDefaults()
+	if importOptions.importRecordsFunction == nil {
+		// if import records function was not already set, set it.
+		if field.options != nil && field.options.fieldType == FieldTypeInt {
+			importRecordsFunction(c.importValues)(&importOptions)
+		} else {
+			// Check whether roaring imports is available
+			if importOptions.wantRoaring {
+				importOptions.hasRoaring = c.hasRoaringImportSupport(field)
+			}
+			importRecordsFunction(c.importColumns)(&importOptions)
+		}
+	}
+	return c.importManager.Run(field, iterator, importOptions)
 }
 
-func (c *Client) importColumns(field *Field, shard uint64, records []Record, nodes []fragmentNode, options *ImportOptions) error {
+func (c *Client) importColumns(field *Field,
+	shard uint64,
+	records []Record,
+	nodes []fragmentNode,
+	options *ImportOptions,
+	state *importState) error {
 	eg := errgroup.Group{}
 
 	if len(nodes) == 0 {
 		return errors.New("No nodes to import to")
 	}
 
-	if options.roaring {
+	if options.hasRoaring {
```

Comment:
This is starting to look large/complicated enough to be a separate function, maybe?

Reply:
thanks, updated